src/commands/moderation/move.js

```diff
@@ -0,0 +1,157 @@
+const { Command, Constants, SwitchbladeEmbed } = require('../../')
+
+module.exports = class Move extends Command {
+  constructor (client) {
+    super({
+      name: 'move',
+      aliases: ['mover'],
+      category: 'moderation',
+      requirements: { guildOnly: true, botPermissions: ['MANAGE_MESSAGES'], permissions: ['MANAGE_MESSAGES'] },
+      parameters: [{
+        type: 'url', missingError: 'commands:move.missingMessageLink'
+      }, {
+        type: 'channel', onlySameGuild: true, acceptText: true, missingError: 'commands:move.missingChannelMention'
+      }]
+    }, client)
+  }
+
+  async run ({ channel, guild, author, t, message }, link, destinationChannel) {
+    /*
+    Bot has manage messages and user also has,
+    so now we need to check those permissions in
+    the channel he wants to move to.
+    */
+    const matchedLink = link.pathname.length > 1 ? link.pathname.match(/channels\/([0-9]{16,18})\/([0-9]{16,18})\/([0-9]{16,18})/) : null
+    const embed = new SwitchbladeEmbed(author)
+    if (!matchedLink) {
+      embed.setTitle(t('commands:move.invalidMessageLink'))
+      embed.setDescription(t('commands:move.missingMessageLink'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    const guildId = matchedLink[1]
+    const channelId = matchedLink[2]
+    const messageId = matchedLink[3]
+
+    if (guildId !== guild.id) {
+      embed.setDescription(t('commands:move.onlyThisGuild'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!guild.channels.has(channelId)) {
+      embed.setDescription(t('commands:move.channelDoesntExist'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    const channelFrom = guild.channels.get(channelId)
+
+    if (!channelFrom) {
+      embed.setDescription(t('commands:move.couldntFindChannel'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
+
+    if (!destinationChannel.permissionsFor(this.client.user.id).has('SEND_MESSAGES')) {
+      embed.setDescription(t('commands:move.cantSendInChannel'))
+      embed.setColor(Constants.ERROR_COLOR)
+      channel.send(embed)
+      return
+    }
```

Comment:
How about turning this into `userChannelPermissions` and `botChannelPermissions` property on the `channel` parameter?